src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -185,6 +185,15 @@ namespace QgsWfs
       }
     }
 
+    // check if all typename are valid
+    for ( const QString &typeName : typeNameList )
+    {
+      if ( !mapLayerMap.contains( typeName ) )
+      {
+        throw QgsRequestNotWellFormedException( QStringLiteral( "TypeName '%1' unknown" ).arg( typeName ) );
```

Comment:
Maybe something a little more verbose? Like "TypeName '%1' could not be found."

Reply:
Done